src/ButteredBread.sol

```diff
@@ -0,0 +1,168 @@
+    }
+
+    /**
+     * @notice Deposit LP tokens
+     * @param _lp Liquidity Pool token
+     * @param _amount Value of LP token
+     */
+    function deposit(address _lp, uint256 _amount) external onlyAllowed(_lp) {
+        _deposit(msg.sender, _lp, _amount);
+    }
+
+    /**
+     * @notice Withdraw LP tokens
+     * @param _lp Liquidity Pool token
+     * @param _amount Value of LP token
+     */
+    function withdraw(address _lp, uint256 _amount) external onlyAllowed(_lp) {
+        _withdraw(msg.sender, _lp, _amount);
+    }
+
+    /**
+     * @notice Allow or deny LP token
+     * @param _lp Liquidity Pool token
+     * @param _allowed Sanction status of LP token
+     * Note: Must set scaling factor before sanctioning LP token
+     */
+    function modifyAllowList(address _lp, bool _allowed) external onlyOwner {
+        if (scalingFactors[_lp] < FIXED_POINT_PERCENT) revert Unset();
+        allowlistedLPs[_lp] = _allowed;
+    }
+
+    /**
+     * @notice Set LP token scaling factor
+     * @param _lp Liquidity Pool token
+     * @param _factor Scaling percentage incentive of LP token (e.g. 100 = 1X, 150 = 1.5X, 1000 = 10X)
+     */
+    function modifyScalingFactor(address _lp, uint256 _factor, address[] calldata holders) external onlyOwner {
+        _modifyScalingFactor(_lp, _factor, holders);
+    }
+
+    /// @notice ButteredBread tokens are non-transferable
+    function transfer(address, uint256) public virtual override returns (bool) {
+        revert NonTransferable();
+    }
+
+    /// @notice ButteredBread tokens are non-transferable
+    function transferFrom(address, address, uint256) public virtual override returns (bool) {
+        revert NonTransferable();
+    }
+
+    /// @notice Deposit LP tokens and mint ButteredBread with corresponding LP scaling factor
+    function _deposit(address _account, address _lp, uint256 _amount) internal {
+        IERC20(_lp).transferFrom(_account, address(this), _amount);
+        _accountToLPData[_account][_lp].balance += _amount;
+
+        uint256 currentScalingFactor = scalingFactors[_lp];
+        _accountToLPData[_account][_lp].scalingFactor = currentScalingFactor;
+
+        _mint(_account, _amount * currentScalingFactor / FIXED_POINT_PERCENT);
+        if (this.delegates(_account) == address(0)) _delegate(_account, _account);
+
+        emit AddButter(_account, _lp, _amount);
+    }
+
+    /// @notice Withdraw LP tokens and burn ButteredBread with corresponding LP scaling factor
+    function _withdraw(address _account, address _lp, uint256 _amount) internal {
+        if (_amount > _accountToLPData[_account][_lp].balance) revert InsufficientFunds();
+
+        /// @dev ensure proper accounting in case of admin error in `modifyScalingFactor` where not all holders are updated
+        _syncVotingWeight(_account, _lp);
+        _accountToLPData[_account][_lp].balance -= _amount;
+
+        _burn(_account, _amount * scalingFactors[_lp] / FIXED_POINT_PERCENT);
```

Comment:
`(_amount * scalingFactors[_lp]) / FIXED_POINT_PERCENT` would make order of operations more clear